http/src/main/resources/jupyter/gce-init.sh

```diff
@@ -0,0 +1,528 @@
+
+    log 'Copying secrets from GCS...'
+
+    mkdir /work
+    mkdir /certs
+    chmod a+rwx /work
+
+    # Add the certificates from the bucket to the VM. They are used by the docker-compose file
+    gsutil cp ${SERVER_CRT} /certs
+    gsutil cp ${SERVER_KEY} /certs
+    gsutil cp ${ROOT_CA} /certs
+    gsutil cp ${PROXY_SITE_CONF} /etc
+    gsutil cp ${JUPYTER_DOCKER_COMPOSE} /etc
+    gsutil cp ${RSTUDIO_DOCKER_COMPOSE} /etc
+    gsutil cp ${PROXY_DOCKER_COMPOSE} /etc
+    gsutil cp ${WELDER_DOCKER_COMPOSE} /etc
+
+    # Needed because docker-compose can't handle symlinks
+    touch /hadoop_gcs_connector_metadata_cache
+
+    # Not all images have the directory used for Stackdriver configs. If so, create it
+    mkdir -p /etc/google-fluentd/config.d
+
+    # Add stack driver configuration for welder
+    tee /etc/google-fluentd/config.d/welder.conf << END
+<source>
+ @type tail
+ format json
+ path /work/welder.log
+ pos_file /var/tmp/fluentd.welder.pos
+ read_from_head true
+ tag welder
+</source>
+END
+
+    # Add stack driver configuration for jupyter
+    tee /etc/google-fluentd/config.d/jupyter.conf << END
+<source>
+ @type tail
+ format none
+ path /work/jupyter.log
+ pos_file /var/tmp/fluentd.jupyter.pos
+ read_from_head true
+ tag jupyter
+</source>
+END
+
+    # Add stack driver configuration for user startup and shutdown scripts
+    tee /etc/google-fluentd/config.d/daemon.conf << END
+<source>
+ @type tail
+ format none
+ path /var/log/daemon.log
+ pos_file /var/tmp/fluentd.google.user.daemon.pos
+ read_from_head true
+ tag daemon
+</source>
+END
+
+    service google-fluentd reload
+    
+    # Install env var config
+    if [ ! -z ${CUSTOM_ENV_VARS_CONFIG_URI} ] ; then
+      log 'Copy custom env vars config...'
+      gsutil cp ${CUSTOM_ENV_VARS_CONFIG_URI} /etc
+    fi
+
+    if [ ! -z ${SERVICE_ACCOUNT_CREDENTIALS} ] ; then
+      echo "GOOGLE_APPLICATION_CREDENTIALS=/etc/${SERVICE_ACCOUNT_CREDENTIALS}" > /etc/google_application_credentials.env
+    else
+      echo "" > /etc/google_application_credentials.env
+    fi
+
+    # If any image is hosted in a GCR registry (detected by regex) then
+    # authorize docker to interact with gcr.io.
+    if grep -qF "gcr.io" <<< "${JUPYTER_DOCKER_IMAGE}${RSTUDIO_DOCKER_IMAGE}${PROXY_DOCKER_IMAGE}${WELDER_DOCKER_IMAGE}" ; then
+      log 'Authorizing GCR...'
+      gcloud --quiet auth configure-docker
+    fi
+
+    STEP_TIMINGS+=($(date +%s))
+
+    log 'Starting up the Jupydocker...'
+
+    # Run docker-compose for each specified compose file.
+    # Note the `docker-compose pull` is retried to avoid intermittent network errors, but
+    # `docker-compose up` is not retried.
+    COMPOSE_FILES=(-f /etc/`basename ${PROXY_DOCKER_COMPOSE}`)
+    cat /etc/`basename ${PROXY_DOCKER_COMPOSE}`
+    if [ ! -z ${JUPYTER_DOCKER_IMAGE} ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${JUPYTER_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${JUPYTER_DOCKER_COMPOSE}`
```

Comment:
Do we need a GCE version of `jupyter-docker-compose.yml`? I think the existing one tries to mount in some Hadoop stuff. Or does it work as is?

Reply:
Good question. It worked as is since the directories seem to exist on GCE too but it probably makes sense to create a GCE-specific `yml` since we don't need any of those volumes there.

Reply:
Yeah I'd probably lean towards a GCE-specific docker-compose yaml